docs/pages/machine-id/reference/configuration.mdx

```diff
@@ -0,0 +1,63 @@
+---
+title: Machine ID Configuration Reference
+description: Configuration reference for Teleport Machine ID.
+---
+
+```yaml
+# Debug enables verbose logging to stderr.
+debug: true
+
+# Address of the Teleport Auth Server (on-prem installs) or Proxy Server
+# (Cloud installs).
+auth_server: "auth.example.com:3025" (On-Prem)
+auth_server: "example.teleport.sh:443" (Cloud)
+
+# TTL of short-lived machine certificates.
+certificate_ttl: "5m"
+
+# Interval at which short-lived certificates are renewed; must be less than
+# the certificate TTL.
+renewal_interval: "1m"
+
+# If set, quit after the first renewal.                                                          |
+oneshot: true
+
+# Onboarding values only used on first connect.
+onboarding:
+    # Cluster join method. Can be "token" or "iam".
+    join_method: "token"
+    # Token used to join cluster.
+    token: "00000000000000000000000000000000"
+    // CA Pin used to validate the identity of Teleport Auth Server on first connect.
+    ca_pin: "sha256:1111111111111111111111111111111111111111111111111111111111111111"
+    // CA Pins used to validate the identity of Teleport Auth Server on first connect.
+    ca_pins:
+    - "sha256:1111111111111111111111111111111111111111111111111111111111111111"
+    - "sha256:2222222222222222222222222222222222222222222222222222222222222222"
+
+# Storage defines where Machine ID internal data is stored.
+storage:
+  # Directory to store internal bot data. Access to this directory should be
+  # limited.
+  directory: /var/lib/teleport/bot
+
+# Destinations specify one or more impersonated certificates sets to generate.
+destinations:
+  # `directory` is the only destination supported in the initial iteration;
+  # the tbot process must be given permissions to write to this directory.
+  - directory: /opt/machine-id
+    # One of more roles to grant to the bot. It must have been granted (at
+    # least) these roles with `tctl bots add --roles=...`
+    # By default, all possible roles are included.
+    roles: [a, b, c]
+    # Which types of certificates to generate. `ssh` is the default.
+    kinds: [ssh]
+    # A list of configuration templates to generate and write to the
+    # destination. Defaults are decided based on the certificate kinds:
+    # `ssh` implies `ssh-client` unless manually (not) configured.
+    configs:
+      # ssh-client generates known_hosts and an ssh_config that can be
+      # included. We can ensure the correct certificate kinds are generated
+      # while generating the config templates.
+      - ssh-client
```

Comment:
Tim: I pulled this straight from the RFD. Is this still accurate?

Reply:
There's only one small error in the above (will need to update the RFD!): `ssh-client` should be `ssh_client`.  That example is otherwise valid but no longer exhaustive.

`directory` now has a verbose form that lets you configure ACL and symlink enforcement:
```yaml
directory:
  path: /opt/machine-id

  # Configure symlink attack prevention. Requires Linux 5.6+.
  # Possible values:
  #   try-secure (default): attempt to securely read and write certificates without symlinks, but fall back (with a warning) to insecure read and write if the host doesn't support it
  #   secure: attempt to securely read and write certificates, hard error if unsupported
  #   insecure: quietly allow symlinks in paths
  symlinks: try-secure

  # Configure ACL use. Requires Linux with a filesystem that supports ACLs.
  # Possible values:
  #   try (default on Linux): attempt to use ACLs, warn at runtime if ACLs are configured but invalid
  #   off (default on non-Linux): do not attempt to use ACLs
  #   required: always use ACLs, produce a hard error at runtime if ACLs are invalid
  acls: try
```

Both are also valid for `storage.directory` entries but only `symlinks` has an effect there. We don't warn about or enforce ACLs at runtime for the bot's internal storage (at least not yet).

The `ssh_client` template also has a verbose form that allows you to override the Teleport proxy port if it's using a non-standard value:
```yaml
configs:
  # ssh-client generates known_hosts and an ssh_config that can be
  # included. We can ensure the correct certificate kinds are generated
  # while generating the config templates.
  - ssh_client:
      proxy_port: 1234

```